editors/code/src/scopes_mapper.ts

```diff
@@ -0,0 +1,79 @@
+import * as vscode from 'vscode';
+import { TextMateRuleSettings } from './scopes';
+
+let mappings = new Map<string, string[]>();
+
+const defaultMapping = new Map<string, string[]>([
+    [
+        'comment',
+        [
+            'comment',
+            'comment.block',
+            'comment.line',
+            'comment.block.documentation'
+        ]
+    ],
+    ['string', ['string']],
+    ['keyword', ['keyword']],
+    ['keyword.control', ['keyword.control', 'keyword', 'keyword.other']],
+    [
+        'keyword.unsafe',
+        ['storage.modifier', 'keyword.other', 'keyword.control', 'keyword']
+    ],
+    ['function', ['entity.name.function']],
+    ['parameter', ['variable.parameter']],
+    ['constant', ['constant', 'variable']],
+    ['type', ['entity.name.type']],
+    ['builtin', ['variable.language', 'support.type', 'support.type']],
+    ['text', ['string', 'string.quoted', 'string.regexp']],
+    ['attribute', ['keyword']],
+    ['literal', ['string', 'string.quoted', 'string.regexp']],
+    ['macro', ['support.other']],
+    ['variable', ['variable']],
+    ['variable.mut', ['variable', 'storage.modifier']],
+    [
+        'field',
+        [
+            'variable.object.property',
+            'meta.field.declaration',
```

Comment:
I think it was discussed before, but `meta` here seems strange.

> For example the entire line which declares a function would be meta.function and the subsets would be storage.type, entity.name.function, variable.parameter etc. and only the latter would be styled. Sometimes the meta part of the scope will be used only to limit the more general element that is styled, most of the time meta scopes are however used in scope selectors for activation of bundle items

Reply:
@lnicola I agree, regarding the defaults, I'd like some solid suggestions from you guys, I just picked what I noticed other themes were using when inspecting TM scopes.

Reply:
Yeah, sorry. I still have to give it a try and think a bit about the mapping we want.